File: ios/FilmRecommend/Models/Movie.swift
import Foundation

struct Movie {
    let index: Int
    let title: String
    let genres: [String]
    let keywords: [String]
    let directorName: String
    let actor1: String
    let actor2: String
    let actor3: String
    let imdbScore: Double
    let numVotedUsers: Int
    let titleYear: Int
}

struct Recommendation: Identifiable {
    let id = UUID()
    let title: String
    let year: Int
    let imdbScore: Double
    let score: Double
    let posterPath: String?
    let trailerKey: String?
    let originalLanguage: String

    init(title: String, year: Int, imdbScore: Double, score: Double, posterPath: String? = nil, trailerKey: String? = nil, originalLanguage: String = "en") {
        self.title = title
        self.year = year
        self.imdbScore = imdbScore
        self.score = score
        self.posterPath = posterPath
        self.trailerKey = trailerKey
        self.originalLanguage = originalLanguage
    }
}

File: tablet/app/src/main/java/com/sanjaybhat/filmrecommend/tablet/model/Movie.kt
package com.sanjaybhat.filmrecommend.tablet.model

data class Movie(
    val index: Int,
    val title: String,
    val genres: List<String>,
    val keywords: List<String>,
    val directorName: String,
    val actor1: String,
    val actor2: String,
    val actor3: String,
    val imdbScore: Double,
    val numVotedUsers: Int,
    val titleYear: Int
)

data class Recommendation(
    val title: String,
    val year: Int,
    val imdbScore: Double,
    val score: Double,
    val posterPath: String? = null,
    val trailerKey: String? = null,
    val originalLanguage: String = "en"
)
